src/EnergyPlus/General.cc

```diff
@@ -333,10 +513,23 @@ namespace General {
 			DY = Y0 - Y1;
 			if ( std::abs( DY ) < SMALL ) DY = SMALL;
 			// new estimation
-			XTemp = ( Y0 * X1 - Y1 * X0 ) / DY;
+			if ( ( HVACSystemRootFinding.TypeNum == Bisection ) ||
+				( HVACSystemRootFinding.TypeNum == Alternation && AltIte > HVACSystemRootFinding.NumOfIter ) ||
+				( HVACSystemRootFinding.TypeNum == RegulaFalsiThenBisection && NIte > HVACSystemRootFinding.NumOfIter ) ||
+				( HVACSystemRootFinding.TypeNum == BisectionThenRegulaFalsi && NIte <= HVACSystemRootFinding.NumOfIter ) ) {
+				if ( HVACSystemRootFinding.TypeNum == Alternation ) {
+					if ( AltIte >= 2 * HVACSystemRootFinding.NumOfIter ) AltIte = 0;
+				}
+				// Bisection
+				XTemp = ( X1 + X0 ) / 2.0;
+			} else {
+				// Regula Falsi
+				XTemp = ( Y0 * X1 - Y1 * X0 ) / DY;
+			}
```

Comment:
@lgu1234 An `enum` is not required, but it does make things more clear (in my opinion).

If there's fall-through in the switch, it'd best to include a comment so that it is clear that is the intent. Some compilers/IDEs/checkers are likely to complain.